### PR TITLE
DP-768: Shared user sessions ADR

### PR DESCRIPTION
**Pull request by mareg**

### CODE REVIEW

File: docs/development/decisions/0007-shared-sessions.adoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+Advantages:
+
+* Easy to set up
+* No code changes required
+* No additional cost
+
+Disadvantages:
+
+* User's journeys can be disrupted when container restarts or new version is deployed
+* Sub-optimal user distribution per running containers
```
Comment (webit4me) [inline docs/development/decisions/0007-shared-sessions.adoc:35]:
```suggestion
* User journeys can be disrupted by sub-optimal distribution across containers, especially during container restarts, i.e. deployments.
```

File: docs/development/decisions/0007-shared-sessions.adoc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+Advantages:
+
+* Serverless - easy to scale up or down depending on demand
+* Easy to implement
+* Low latency - can provide session details with a minimal lag
+* Can set up TTL, so expired sessions will be automatically removed
+
+Disadvantages:
+
+* Slightly more expensive than DynamoDB
```
Comment (webit4me) [inline docs/development/decisions/0007-shared-sessions.adoc:60]: Not knowing the exact requirement and load on this component as described in the context section, this statement can be incorrect.

**Review by webit4me — APPROVED** [review]

File: docs/development/decisions/0007-shared-sessions.adoc [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+Advantages:
+
+* Easy to set up
+* No code changes required
+* No additional cost
+
+Disadvantages:
+
+* User journeys can be disrupted by sub-optimal distribution across containers, especially during container restarts, i.e. deployments.
```
Comment (jakzal) [inline docs/development/decisions/0007-shared-sessions.adoc:34]: One more disadvantage I see:
* sessions are destroyed when containers restart (i.e. during a release)

Comment (webit4me) [reply docs/development/decisions/0007-shared-sessions.adoc:34]: That is the reason of the user journey disruption, not a second reason. perhaps the sentence can be improved to  ...
User journeys can be disrupted by sub-optimal user journey distribution across containers when container restarts, i.e. deployments.

but also worth to clarify if its for all containers or there are individual containers in charge of managing the session.

**Comment by dharmverma** [comment]
Redis, good choice in my opinion

**Review by dharmverma — APPROVED** [review]

**Review by rmohammed-goaco — APPROVED** [review]
Thumbs up from me

**Review by jakzal — APPROVED** [review]

**Pull request merged**